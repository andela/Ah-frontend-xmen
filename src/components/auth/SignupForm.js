/* eslint-disable jsx-a11y/anchor-is-valid */

import React from 'react';
import SocialAuthButtons from '../../views/socialAuthView/socialAuthView';

const SignupForm = props => (
  <div className="card-body px-lg-3  py-lg-3">
    <div className="text-center">
      <h3 className="text-primary">Signup with</h3>
      <SocialAuthButtons onSuccess={props.onSuccess} />
    </div>
    <div className="row p-3">
      <span className="col-md-5"><hr /></span>
      <span className="col-md-2 text-center">or</span>
      <span className="col-md-5"><hr /></span>
    </div>
    <form onSubmit={props.onSubmit} id="signup-form">
      <div className="form-group mb-3">
        <div className="input-group input-group-alternative">
          <div className="input-group-prepend">
            <span className="input-group-text"><i className="fa fa-envelope" /></span>
          </div>
          <input className="form-control" required onChange={props.onChange} placeholder="Email" type="email" id="email" />
        </div>
      </div>
      <div className="form-group">
        <div className="input-group input-group-alternative">
          <div className="input-group-prepend">
            <span className="input-group-text"><i className="fa fa-user" /></span>
          </div>
          <input className="form-control" required onChange={props.onChange} placeholder="Username" type="text" id="username" />
        </div>
      </div>
      <div className="form-group">
        <div className="input-group input-group-alternative">
          <div className="input-group-prepend">
            <span className="input-group-text"><i className="fa fa-lock" /></span>
          </div>
          <input className="form-control" required onChange={props.onChange} id="password" placeholder="Password" type="password" />
        </div>
      </div>
      <div className="text-center">
        <input type="submit" className="btn btn-block button-primary my-4" value="Sign Up" />
      </div>
      <div className="text-center">
Already have an account,
        <a href="#" onClick={props.onLoginTrigger}> Login</a>

      </div>
    </form>
  </div>
);


export default SignupForm;
