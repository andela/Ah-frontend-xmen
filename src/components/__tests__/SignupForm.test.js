import { shallow } from 'enzyme';
import React from 'react';
import SignupForm from '../auth/SignupForm';

describe('Signupform', () => {
  it('Should render without crashing', () => {
    const wrapper = shallow(<SignupForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
