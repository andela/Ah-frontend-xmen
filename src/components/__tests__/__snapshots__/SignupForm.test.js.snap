// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signupform Should render without crashing 1`] = `ShallowWrapper {}`;
